File: debug_lexer_test.rs
use cursed::lexer::Lexer;

fn main() {
    let source = "based lies lowkey";
    let mut lexer = Lexer::new(source.to_string());
    
    while let Ok(token) = lexer.next_token() {
        if token.kind == cursed::lexer::TokenKind::Eof {
            break;
        }
        println!("Token: {:?} - '{}'", token.kind, token.lexeme);
    }
}

File: debug_lexer_test.zig
const std = @import("std");
const lexer = @import("src-zig/lexer.zig");

pub fn main() !void {
    const allocator = std.heap.page_allocator;
    
    const source = "damn 42;";
    var lex = lexer.Lexer.init(allocator, source);
    const tokens = try lex.tokenize();
    
    for (tokens.items) |token| {
        std.debug.print("Token: {s}, Kind: {any}\n", .{ token.lexeme, token.kind });
    }
}
